server/market/orderrouter.go

```diff
@@ -167,6 +176,12 @@ func (r *OrderRouter) handleLimit(user account.AccountID, msg *msgjson.Message)
 		return rpcErr
 	}
 
+	// Spare some resources if the market is closed now. Any orders that make it
+	// through to a closed market will receive a similar error from SubmitOrder.
+	if !tunnel.Running() {
+		return msgjson.NewError(msgjson.MarketNotRunningError, "market closed to new orders")
+	}
```

Comment:
In the present design, the order router commanded the suspension and thus can track the status of each market, at least nominally, so this will likely not query the MarketTunnel in the future, but `handleLimit` should catch orders submitted to suspended markets asap to save resources rather than sending them all the way to SubmitOrder to be bounced.